io/io/src/TBufferFile.cxx

```diff
@@ -2015,10 +2050,18 @@ void TBufferFile::WriteFastArray(const Short_t *h, Int_t n)
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Write array of n ints into the I/O buffer.
+/// \note Operation throws error in case of underflow or overflow
```

Comment:
```suggestion
/// \warning Due to the current limit of the buffer size, the function aborts execution of the program in case of underflow or overflow. See https://github.com/root-project/root/issues/6734 for more details.
```

And similarly in other methods